contracts/Ops.sol

```diff
@@ -1,292 +1,291 @@
-    /// @param _execData Data used to execute the tx, queried from the Resolver by Gelato
-    // solhint-disable function-max-lines
-    // solhint-disable code-complexity
+    // prettier-ignore
+    fallback(bytes calldata _callData) external returns(bytes memory returnData){
+        returnData = _handleLegacyTaskCreation(_callData);
+    }
+
+    ///@inheritdoc IOps
+    function createTask(
+        address _execAddress,
+        bytes calldata _execData,
+        LibDataTypes.ModuleData calldata _moduleData,
+        address _feeToken
+    ) external override returns (bytes32 taskId) {
+        taskId = _createTask(
+            msg.sender,
+            _execAddress,
+            _execData,
+            _moduleData,
+            _feeToken
+        );
+    }
+
+    ///@inheritdoc IOps
+    function cancelTask(bytes32 _taskId) external {
+        _cancelTask(msg.sender, _taskId);
+    }
+
+    ///@inheritdoc IOps
     function exec(
+        address _taskCreator,
+        address _execAddress,
+        bytes memory _execData,
+        LibDataTypes.ModuleData calldata _moduleData,
+        uint256 _txFee,
+        address _feeToken,
+        bool _useTaskTreasuryFunds,
+        bool _revertOnFailure
+    ) external onlyGelato {
+        bytes32 taskId = LibTaskId.getTaskId(
+            _taskCreator,
+            _execAddress,
+            _execData.memorySliceSelector(),
+            _moduleData,
+            _useTaskTreasuryFunds ? address(0) : _feeToken
+        );
+
+        _exec(
+            taskId,
+            _taskCreator,
+            _execAddress,
+            _execData,
+            _moduleData.modules,
+            _txFee,
+            _feeToken,
+            _useTaskTreasuryFunds,
+            _revertOnFailure
+        );
+    }
+
+    ///@inheritdoc IOps
+    function legacyExec(
         uint256 _txFee,
         address _feeToken,
         address _taskCreator,
         bool _useTaskTreasuryFunds,
         bool _revertOnFailure,
         bytes32 _resolverHash,
         address _execAddress,
-        bytes calldata _execData
-    ) external override onlyGelato {
-        bytes32 taskId = getTaskId(
+        bytes memory _execData
+    ) external onlyGelato {
+        bytes32 taskId = LibTaskId.getLegacyTaskId(
             _taskCreator,
             _execAddress,
-            _execData.calldataSliceSelector(),
+            _execData.memorySliceSelector(),
             _useTaskTreasuryFunds,
             _useTaskTreasuryFunds ? address(0) : _feeToken,
             _resolverHash
         );
 
-        require(
-            _taskCreator != address(0) && taskCreator[taskId] == _taskCreator,
-            "Ops: exec: No task found"
-        );
-
-        if (!_useTaskTreasuryFunds) {
-            fee = _txFee;
-            feeToken = _feeToken;
-        }
+        LibDataTypes.Module[] memory modules = new LibDataTypes.Module[](2);
+        modules[0] = LibDataTypes.Module.RESOLVER;
+        modules[1] = LibDataTypes.Module.TIME;
```

Comment:
couldnt we just do this check in here: 

```
Time storage time = timedTask[task];
bool isTimedTask = time.nextExec != 0 ? true : false;
```
and then if `isTimeTask = true` we add the TIME module? Otherwise we add the extra gas overhead of delegate calling into the TimeModule for every legacy task